apps/explorer/src/pages/object-result/views/ObjectView.tsx

```diff
@@ -0,0 +1,224 @@
+import { Heading, Text } from '@mysten/ui';
+import { type ReactNode, useEffect, useState } from 'react';
+
+import { useResolveVideo } from '~/hooks/useResolveVideo';
+import { Card } from '~/ui/Card';
+import { DescriptionItem, type DescriptionItemProps } from '~/ui/DescriptionList';
+import { AddressLink, ObjectLink, TransactionLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { ObjectVideoImage } from '~/ui/ObjectVideoImage';
+import { extractName, getDisplayUrl, parseImageURL, parseObjectType } from '~/utils/objectUtils';
+import { genFileTypeMsg, trimStdLibPrefix } from '~/utils/stringUtils';
+
+interface HeroVideoImageProps {
+	title: string;
+	subtitle: string;
+	src: string;
+	variant: 'xl';
+	video?: string | null;
+}
+
+function HeroVideoImage({ title, subtitle, src, video, variant }: HeroVideoImageProps) {
+	const [open, setOpen] = useState(false);
+
+	return (
+		<div className="group relative">
+			<ObjectVideoImage
+				title={title}
+				subtitle={subtitle}
+				src={src}
+				video={video}
+				variant={variant}
+				open={open}
+				setOpen={setOpen}
+				rounded="xl"
+			/>
+			<ArrowUpRight16 className="right-5 top-5 hidden group-hover:absolute group-hover:block" />
+		</div>
+	);
+}
+
+function ObjectViewItem({
+	title,
+	align,
+	children,
+}: {
+	title: string;
+	children: ReactNode;
+	align?: DescriptionItemProps['align'];
+}) {
+	return (
+		<DescriptionItem
+			descriptionJustify="end"
+			labelWidth="lg"
+			align={align}
+			title={
+				<Text variant="pBodySmall/medium" color="steel-dark">
+					{title}
+				</Text>
+			}
+		>
+			{children}
+		</DescriptionItem>
+	);
+}
+
+function ObjectViewCard({ children }: { children: ReactNode }) {
+	return (
+		<div>
+			<Card bg="white/80" spacing="lg" height="full">
+				<div className="flex flex-col gap-4">{children}</div>
+			</Card>
+		</div>
+	);
+}
+
+function LinkWebsite({ value }: { value: string }) {
+	const urlData = getDisplayUrl(value);
+
+	if (!urlData) {
+		return null;
+	}
+
+	if (typeof urlData === 'string') {
+		return <Text variant="pBodySmall/medium">{urlData}</Text>;
+	}
+
+	return (
+		<Link href={urlData.href} variant="textHeroDark">
+			{urlData.display}
+		</Link>
+	);
+}
+
+interface ObjectViewProps {
+	data: SuiObjectResponse;
+}
+export function ObjectView({ data }: ObjectViewProps) {
+	const [fileType, setFileType] = useState<undefined | string>(undefined);
+	const display = data.data?.display?.data;
+	const imgUrl = parseImageURL(display);
+	const video = useResolveVideo(data);
+	const name = extractName(display);
+	const objectType = parseObjectType(data);
+	const objOwner = data.data?.owner;
+	const storageRebate = data.data?.storageRebate;
+
+	const [storageRebateFormatted] = useFormatCoin(storageRebate, SUI_TYPE_ARG, CoinFormat.FULL);
+
+	const { address, module } = parseStructTag(objectType);
+
+	useEffect(() => {
+		const controller = new AbortController();
+		genFileTypeMsg(imgUrl, controller.signal)
```

Comment:
If we're rewriting this, can we also rewrite this to either use `useQuery` or avoid this altogether or something

Reply:
updated to `useQuery`